core/modules/user/user.pages.inc

```diff
@@ -516,16 +537,30 @@ function user_cancel_methods() {
 
   // Turn all methods into real form elements.
   $default_method = $site_config->get('user_cancel_method');
+  $options = array();
   foreach ($methods as $name => $method) {
-    $form[$name] = array(
-      '#type' => 'radio',
-      '#title' => $method['title'],
-      '#description' => (isset($method['description']) ? $method['description'] : NULL),
-      '#return_value' => $name,
-      '#default_value' => $default_method,
-      '#parents' => array('user_cancel_method'),
-    );
+    $options[$name] = $method['title'];
+    // Flag used to denote account cancellation methods as being
+    // dangerous/irreversible.
+    if (isset($method['danger']) && $method['danger']) {
+      $additional_properties[$name]['#danger'] = $method['danger'];
+    }
+    $additional_properties[$name]['#description'] = $method['description'];
+    // Flag used to exclude cancellation methods from being set as the
+    // default.
+    if (isset($method['access']) && $method['access']) {
+      $additional_properties[$name]['#access'] = $method['access'];
+    }
```

Comment:
This is a preexisting property of each cancellation option. What I'm doing here is to just make sure that it is actually available for any function that happens to call `user_cancel_methods()`. What we actually do with it in `user_admin_settings()` (`admin/config/people/settings`) is to be decided in https://github.com/backdrop/backdrop-issues/issues/5121